src/javascript/contexts/ContentEditorSection/ContentEditorSection.context.jsx

```diff
@@ -1,20 +1,39 @@
-import React, {useContext, useRef, useState} from 'react';
+import React, {useContext, useEffect, useRef, useState} from 'react';
 import * as PropTypes from 'prop-types';
 
 export const ContentEditorSectionContext = React.createContext({});
 
 export const useContentEditorSectionContext = () => useContext(ContentEditorSectionContext);
-export const ContentEditorSectionContextProvider = ({formSections, children}) => {
-    const sections = useRef(formSections);
 
+export const ContentEditorSectionContextProvider = ({formSections, children}) => {
+    const previousValue = useRef();
+    const [sections, setSectionsState] = useState([]);
+    const [expanded, setExpanded] = useState({});
     const [, setChangeCount] = useState(0);
 
+    const stringifiedSections = formSections && JSON.stringify(formSections);
+    if (stringifiedSections && previousValue.current !== stringifiedSections) {
+        previousValue.current = stringifiedSections;
+        const sectionsCopy = JSON.parse(stringifiedSections);
+        setSectionsState(sectionsCopy);
+        if (Object.keys(expanded).length === 0) {
+            setExpanded(sectionsCopy.reduce((acc, curr) => ({
```

Comment:
`reduce` can be replaced by a `map`

Reply:
how ? unless we convert `expanded` to an array (which could be more simple) ?

Reply:
indeed, the `reduce` is required
```[curr.name]: acc[curr.name] ? acc[curr.name] : curr.expanded``` 
can be simplified